torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1378,6 +1378,46 @@ def _make_tensor_helper(shape, low=None, high=None):
 
     return samples
 
+def sample_inputs_lerp(op_info, device, dtype, requires_grad):
+    def _make_tensor_helper(shape, low=None, high=None):
+        return make_tensor(shape, device, dtype, low=low, high=high, requires_grad=requires_grad)
+
+    samples = (
+        # no broadcast
+        SampleInput((_make_tensor_helper((S, S)), _make_tensor_helper((S, S)), 0.4)),
+        # broadcast rhs
+        SampleInput((_make_tensor_helper((S, S)), _make_tensor_helper((S,)), 0.4)),
+        # scalar tensor
+        SampleInput((_make_tensor_helper(()), _make_tensor_helper(()), 0.4)),
+        # broadcast rhs scalar-tensor
+        SampleInput((_make_tensor_helper((S, S)), _make_tensor_helper(()), 0.4)),
+        # broadcast rhs with weight tensor
+        SampleInput((_make_tensor_helper((S, S)), _make_tensor_helper((S,)), _make_tensor_helper((S, S)))),
+        # broadcast rhs and weight tensor
+        SampleInput((_make_tensor_helper((S, S)), _make_tensor_helper((S, 1)), _make_tensor_helper((S,)))),
+
+        # Broadcasts `self` : Issue with inplace-variants
+        # Reference: https://github.com/pytorch/pytorch/issues/50747
+        # SampleInput((_make_tensor_helper((S,)), _make_tensor_helper((S, S)), 0.4)),
+        # SampleInput((_make_tensor_helper(()), _make_tensor_helper((S, S)), 0.4)),
+        # SampleInput((_make_tensor_helper((S, 1)), _make_tensor_helper((S, S)), 0.4)),
+        # SampleInput((_make_tensor_helper((S, 1)), _make_tensor_helper((S, S)), _make_tensor_helper((S, 1)))),
+    )  # type: ignore
+
+    if dtype.is_complex:
+        samples = samples + (  # type: ignore
+            # no broadcast
+            SampleInput((_make_tensor_helper((S, S)), _make_tensor_helper((S, S)), 0.4j)),
```

Comment:
I think we should test a variation of weight values here, for e.g. weights with non-zero real val as well as floats.

Reply:
We already have case of complex tensor inputs and float weight.

Will add case of complex with non zero real part.